src/backend/common/cache_clearing/get_affected_queries.py

```diff
@@ -0,0 +1,313 @@
+from typing import Any, List, Set, Tuple, Type
+
+from google.cloud import ndb
+
+from backend.common.models.cached_model import TAffectedReferences
+from backend.common.models.district_team import DistrictTeam
+from backend.common.models.event import Event
+from backend.common.models.event_team import EventTeam
+from backend.common.models.keys import TeamKey
+from backend.common.queries import (
+    award_query,
+    district_query,
+    event_details_query,
+    event_query,
+    match_query,
+    media_query,
+    robot_query,
+    team_query,
+)
+from backend.common.queries.database_query import CachedDatabaseQuery
+
+TCacheKeyAndQuery = Tuple[str, Type[CachedDatabaseQuery]]
+
+
+def _queries_to_cache_keys_and_queries(
+    queries: List[CachedDatabaseQuery],
+) -> List[TCacheKeyAndQuery]:
+    out = []
+    for query in queries:
+        out.append((query.cache_key, type(query)))
+    return out
+
+
+def _get_team_page_num(team_key: TeamKey) -> int:
+    return int(team_key[3:]) // team_query.TeamListQuery.PAGE_SIZE
+
+
+def _filter(refs: Set[Any]) -> Set[Any]:
+    # Default filter() filters zeros, so we can't use it.
+    return {r for r in refs if r is not None}
+
+
+def award_updated(affected_refs: TAffectedReferences) -> List[TCacheKeyAndQuery]:
+    event_keys = _filter(affected_refs["event"])
+    team_keys = _filter(affected_refs["team_list"])
+    years = _filter(affected_refs["year"])
+    event_types = _filter(affected_refs["event_type_enum"])
+    award_types = _filter(affected_refs["award_type_enum"])
```

Comment:
Can we make `TAffectedReferences` a bit of a safer class here, where instead of referencing a dictionary we could have these be typed? So `affected_refs.event`, `affected_refs.year`, etc?

Reply:
This would be tricky, because the schema is dependent on the schema of the model. There may be some fancy way to do it, but I'd rather tackle that one separately.